web/packages/core/src/js-polyfills.ts

```diff
@@ -126,6 +127,39 @@ function tryPolyfillReflect(): void {
     }
 }
 
+/**
+ * Replaces a `Map` object missing standard methods with an unchanged `Map` object from a fresh global.
+ *
+ * @returns The custom `Map` object that exists on the page, or undefined if the page uses the standard `Map`.
+ */
+export function resetCustomMap(): typeof Map | undefined {
+    if (typeof Map.prototype.set !== "function") {
+        const currentMap = Map;
+        const iframe = document.createElement("iframe");
+        iframe.style.display = "none";
+        document.documentElement.append(iframe);
+        if (iframe.contentWindow) {
```

Comment:
When is this `null`?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/HTMLIFrameElement/contentWindow#description

Because TS doesn't know if an iframe is cross-origin, it requires checking if `iframe.contentWindow` exists, because in a cross-origin case it may not.

Reply:
I don't know much about TS, but this is where you would use `!` right?